crates/sui-types/src/move_package.rs

```diff
@@ -323,31 +323,41 @@ impl MovePackage {
 
     /// The ObjectID that this package's modules believe they are from, at runtime (can differ from
     /// `MovePackage::id()` in the case of package upgrades).
-    pub fn original_package_id(&self) -> ObjectID {
+    fn original_package_id(&self) -> ObjectID {
         let bytes = self.module_map.values().next().expect("Empty module map");
         let module = CompiledModule::deserialize(bytes)
             .expect("A Move package contains a module that cannot be deserialized");
         (*module.address()).into()
     }
 
-    pub fn deserialize_module(&self, module: &Identifier) -> SuiResult<CompiledModule> {
+    pub fn deserialize_module(
+        &self,
+        module: &Identifier,
+        max_binary_format_version: u32,
+    ) -> SuiResult<CompiledModule> {
         // TODO use the session's cache
         let bytes = self
             .serialized_module_map()
             .get(module.as_str())
             .ok_or_else(|| SuiError::ModuleNotFound {
                 module_name: module.to_string(),
             })?;
-        Ok(CompiledModule::deserialize(bytes)
-            .expect("Unwrap safe because Sui serializes/verifies modules before publishing them"))
+        Ok(
+            CompiledModule::deserialize_with_max_version(bytes, max_binary_format_version).expect(
+                "Unwrap safe because Sui serializes/verifies modules before publishing them",
```

Comment:
Does this message still hold true?  This call can fail when trying to deserialize a module at a bytecode version that is newer than the max, but it's possible that should never happen under normal operation?

Reply:
FWIW expects and other things have lead to crashes in the past, as there were cases where `MovePackage` was used even when the package failed to publish.

Reply:
Changed this to return a deserialization error similarly to what we do in other parts of this package